aws/resource_aws_wafv2_rule_group.go

```diff
@@ -0,0 +1,1361 @@
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"arn": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validateArn,
+				},
+				"field_to_match":      wafv2FieldToMatchSchema(),
+				"text_transformation": wafv2TextTransformationSchema(),
+			},
+		},
+	}
+}
+
+func wafv2SizeConstraintSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"comparison_operator": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						wafv2.ComparisonOperatorEq,
+						wafv2.ComparisonOperatorGe,
+						wafv2.ComparisonOperatorGt,
+						wafv2.ComparisonOperatorLe,
+						wafv2.ComparisonOperatorLt,
+						wafv2.ComparisonOperatorNe,
+					}, false),
+				},
+				"field_to_match": wafv2FieldToMatchSchema(),
+				"size": {
+					Type:         schema.TypeInt,
+					Required:     true,
+					ValidateFunc: validation.IntBetween(0, 21474836480),
+				},
+				"text_transformation": wafv2TextTransformationSchema(),
+			},
+		},
+	}
+}
+
+func wafv2SqliMatchStatementSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"field_to_match":      wafv2FieldToMatchSchema(),
+				"text_transformation": wafv2TextTransformationSchema(),
+			},
+		},
+	}
+}
+
+func wafv2XssMatchStatementSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"field_to_match":      wafv2FieldToMatchSchema(),
+				"text_transformation": wafv2TextTransformationSchema(),
+			},
+		},
+	}
+}
+
+func wafv2FieldToMatchSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"all_query_arguments": wafv2EmptySchema(),
+				"body":                wafv2EmptySchema(),
+				"method":              wafv2EmptySchema(),
+				"query_string":        wafv2EmptySchema(),
+				"single_header": {
+					Type:     schema.TypeList,
+					Optional: true,
+					MaxItems: 1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"name": {
+								Type:     schema.TypeString,
+								Required: true,
+								ValidateFunc: validation.All(
+									validation.StringLenBetween(1, 40),
+									// The value is returned in lower case by the API.
+									// Trying to solve it with StateFunc and/or DiffSuppressFunc resulted in hash problem of the rule field or didn't work.
+									validateWafv2StringIsLowerCase(),
+								),
```

Comment:
though, if the validation is needed here for case, we should use the `validation` package's `StringMatch` like
```
validation.StringMatch(regexp.MustCompile(`^[a-z]+$`), "must contain only lowercase alpha characters"),
```

Reply:
let's keep this validatonfunc but slightly adjust the message to include "lowercase"

Reply:
@pvanbuijtene last note ^^ to adjust the error message that will appear in the output